### PR TITLE
Add democracy-voted extrinsic event and test.

### PR DESCRIPTION
**Pull request by jnaviask**
This adds the `democracy-voted` event to the chain events repo, along with a unit test.

However, this has not been QA'd at all against any chain, so, **before merging, we should verify that these changes are compatible with all supported chains.**

### CODE REVIEW

File: src/substrate/filters/titler.ts [hunk 1]
```diff
@@ -105,6 +105,12 @@ export const Title: TitlerFilter = (kind: EventKind): IEventTitle => {
         description: 'A new democracy referendum started voting.',
       };
     }
+    case EventKind.DemocracyVoted: {
+      return {
+        title: 'Democracy Vote Received',
+        description: 'A democracy vote is received.',
```
Comment (raykyri) [inline src/substrate/filters/titler.ts:111]: This should be past tense 🤷

File: src/substrate/processor.ts [hunk 2]
```diff
@@ -53,7 +53,18 @@ export class Processor extends IEventProcessor<ApiPromise, Block> {
     };
 
     const events = await Promise.all(block.events.map(({ event }) => applyFilters(event)));
-    const extrinsics = await Promise.all(block.extrinsics.map((extrinsic) => applyFilters(extrinsic)));
-    return [...events, ...extrinsics].filter((e) => !!e); // remove null / unwanted events
+
+    // remove unsuccessful extrinsics, only keep extrinsics that map to ExtrinsicSuccess events
+    // cf: https://polkadot.js.org/docs/api/cookbook/blocks#how-do-i-map-extrinsics-to-their-events
+    const successfulExtrinsics = block.extrinsics.filter((_extrinsic, index) => {
```
Comment (raykyri) [inline src/substrate/processor.ts:59]: I don't know how stable or fragile this is...

File: src/substrate/filters/enricher.ts [hunk 3]
```diff
@@ -714,6 +714,25 @@ export async function Enrich(
     excludeAddresses?: string[],
   }> => {
     switch (kind) {
+      case EventKind.DemocracyVoted: {
+        const voter = extrinsic.signer.toString();
+        const [ idx, vote ] = extrinsic.args as [ Compact<ReferendumIndex>, AccountVote ];
+        if (vote.isSplit) {
+          throw new Error('split votes not supported');
```
Comment (raykyri) [inline src/substrate/filters/enricher.ts:721]: Will throwing an error break anything?

**Review by raykyri — APPROVED** [review]
This looks fine to me. Added a couple of comments but I don't think we need to add much more in the way of changes this time around.

**Comment by raykyri** [comment]
Does this still need to be merged?

File: src/substrate/processor.ts (hunk 2 above)
Comment (jnaviask) [reply src/substrate/processor.ts:59]: Quite stable from my testing

File: src/substrate/filters/enricher.ts (hunk 3 above)
Comment (jnaviask) [reply src/substrate/filters/enricher.ts:721]: The pattern in this file is to throw and handle outside of it, so no.

**Pull request merged**